### DISCUSSION TITLE
Importing Preset written in TypeScript for Storybook?

### DISCUSSION THREAD

**Discussion by dance2die** [tailwindlabs/tailwindcss#3667] [Help] [answered]
## What I want to do

I'd like to extend a preset written in TypeScript within `tailwind.config.js` for a Storybook demo in a same project.

## Context

I bootstrapped a new Tailwind CSS ("TW" hereafter) Preset, [downwindcss/easings](https://github.com/downwindcss/easings) ("preset" hereafter) using [TSDX](https://tsdx.io/) (similar to [Microbundle](https://github.com/developit/microbundle) but for TypeScript).

## Issue

The preset is written in TypeScript but `tailwind.config.js` doesn't allow importing a typescript file.
Storybook can use the TW in [this branch](https://github.com/downwindcss/easings/tree/sb) of the source.

Anyone has any suggestion on how to extend the TypeScript preset in `tailwind.config.js`?

![image](https://user-images.githubusercontent.com/8465237/109575961-dd4c3280-7ac0-11eb-9710-d6eea9e616dc.png)

**Comment by simonswiss** [comment, marked as answer]
Hey!

I think you need to require the compiled JS file for your `easing` rather than the TS file directly for it to work.

**Reply by dance2die** [reply]
Thanks @simonswiss.

You are right. I just ended up [importing my published NPM package](https://github.com/downwindcss/easings/blob/sb/tailwind.config.js#L4).

```js
module.exports = {
  purge: [],
  darkMode: false, // or 'media' or 'class'
  presets: [require('@downwindcss/easings')],
};
```
